### DISCUSSION TITLE
Sonoff sv one button controlling the relay and the other sending a mqtt message

### DISCUSSION THREAD

**Discussion by Movieaholic** [arendst/Tasmota#14716] [Q&A] [answered]
Hi I followed this video https://www.youtube.com/watch?v=M76g81G3bi8&list=WL&index=11&t=373s to make a smart door lock controllable by home assistant then a lot later down the track I watched this https://www.youtube.com/watch?v=QMepwpyjMCY&t=6s video and figured out I could use the same device as a contact sensor and Ive had both a button to control the relay and a reed switch as a contact sensor sending mqtt codes working separately but i cant seem to get them working both how i want at the same time, surely its something simple im not sure I totally understand the commands here https://tasmota.github.io/docs/Buttons-and-Switches/ do i have to set a rule?

**Comment by barbudor** [comment, marked as answer]
If you have configured a GPIO as a switch and don't want it to control the relay but send mqtt messages, you have to decouple the switch from the relay.
As stated in the doc you have 4 ways of doing so
- a rule to catch switch state changes and use publish comma d
- switchmode1 15 to decouple switch1
- setoption114 1 to decouple all switches
- swichtopic to have all switches publishing to a given topic.

The 2 in the middle are probably the most easy on Tasmota side but I have no idea how to configure HA to get the masq message
The 1st and 4th options allow you to more finely choose the topic to publish to with the 1st option being the most flexible on that

**Reply by sfromis** [reply]
Maybe the Tasmota docs about HA binary sensor setup can be "inspirational"....
https://tasmota.github.io/docs/Home-Assistant/#binary-sensors

**Comment by Movieaholic** [comment]
Thanks for your help but I have the sensors in home assistant set up and they run the only problem is when i push the switch in the tasmota console is says POWER1 = TOGGLE but it doesnt TOGGLE the relay im guessing its uncoupled already ive put the commands in and set the gpios as in the video

Switchretain 1
Powerretain 1
switchtopic2 garagestate
switchmode1 0
switchmode2 2

GPIO 14 to 10 Switch2

GPIO 4 to 22 Relay2

also set the switch 1 to gpio 5 
can i not have switch 1 coupled to relay 1 and switch 2 uncoupled just sending mqtt comands?
